### PR TITLE
102301121 - work1

### PR DESCRIPTION
**Pull request by WuJunkai2004**
## work1  
Containing the following works
- [x] softmax function
- [x] use PyTorch to recognize the MNIST dataset
- [x] do not use PyTorch to recognize the MNIST dataset

> 最近太摆了，给自己找点动力  
> 太忙了，彻底摆烂了

### CODE REVIEW

**Comment by WuJunkai2004** [comment]
贝叶斯识别对mnist数据集的识别正确率：
```
correct: 5562
wrong: 4438
accuracy: 0.5562
```
应该能够通过改进`清洗`和`处理`的步骤，让正确率进一步提供。

**Review by lyc280705 — COMMENTED** [review]
太强了，快来教我

**Comment by WuJunkai2004** [comment]
拿dataset和dataload重写了一部分代码，修改了一部分代码
目前，贝叶斯的正确率在60%左右
感觉到极限了，周末整理一下，下周去搞pytorch

**Comment by WuJunkai2004** [comment]
基本完成mnist数据集的识别
正确率卡在 88.8%，上不去，也没时间调整了
就先这样吧

**Comment by WuJunkai2004** [comment]
我不理解了
同样的训练集，同样的训练过程，同样的步数，同样的损失函数
我的正确率可以低上三成

脑袋烧掉了

**Review by SYguanxin — APPROVED** [review]
可以啊，虽然贝叶斯我看不懂，写得挺清楚的

**Comment by SYguanxin** [comment]
我第三个要求的意思其实是只用numpy写，但这个也挺好

**Comment by WuJunkai2004** [comment]
我到很晚意识到第三个要求的真正用意是什么。
但是那时候我都快写完了，就没改了
________________________________
发件人: ***@***.*** ***@***.***> 代表 SYguanxin ***@***.***>
发送时间: 2023年12月16日 15:30
收件人: west2-online-reserve/collection-ai ***@***.***>
抄送: Wu Junkai ***@***.***>; Author ***@***.***>
主题: Re: [west2-online-reserve/collection-ai] 102301121 - work1 (PR #1)


我第三个要求的意思其实是只用numpy写，但这个也挺好

―
